### PR TITLE
only remove stale documents an a connect, and skipUpdate for the tracker hook

### PR DESCRIPTION
**Pull request by bratelefant**
If there's a (re)connect, then all docs are removed completely and then readded via initial ddp `added` events. This can result in flashing ui, since items disappear and then reappear after being readded.

I guess this was considered necessary, since during the time with no connection to the server, there may be docs that get removed from the db, and the somehow the client needs to get those docs removed aswell as soon as it is reconnected.

This PR solves this by marking all docs as `_stale` on a disconnect. After a reconnect, docs will get initially added via ddp, and will be flagged as not being `_stale` anymore.

After a certain timeout following a (re)connect (bet there's a better place to do this? if all subs are ready maybe?), all docs, that were removed on the server are still flagged as `_stale`. Now all `_stale` docs can be safely removed from the client.

Also adds the `skipUpdate` callback to the `useTracker` hook, as known from react-meteor-data.

### CODE REVIEW

**Comment by jankapunkt** [comment]
@bratelefant is there something I can do here to help out?

**Comment by bratelefant** [comment]
Actually I'm really not sure weather this is a feature / behavior that is really requested or accepted. That's why I added this as a draft, as sort of a basis for discussion.

**Comment by bratelefant** [comment]
Also, I suppose we need a better way to determine the point when the initial additions of documents are finished to remove stale data; usage of `setTimeout` is kind of messy.

File: companion-packages/meteorrn-local/index.js [hunk 1]
```diff
@@ -96,11 +96,12 @@ const Local = {
 
       LiveCol.find({}).observe({
         added: async (doc) => {
-          LocalCol._collection.upsert(doc);
+          LocalCol.insert(doc);
```
Comment (ToyboxZach) [inline companion-packages/meteorrn-local/index.js:99]: With your _stale items, won't this leave duplicates of all the items in the collection temporarily? This doesn't seem like it makes anything better, just more unpredictable with a timeout

Comment (bratelefant) [reply companion-packages/meteorrn-local/index.js:99]: Yeah, I totally discarded this approach actually. If it's ok I'd rather close this PR, this is not leading to any good. I'm really happy with how #128 is working. Much more predictable results that way.

Comment (ToyboxZach) [reply companion-packages/meteorrn-local/index.js:99]: Yep makes sense, that approach seems to add much fewer edge cases

**Pull request closed without merging**